src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -423,19 +443,27 @@ static grpc_lb_policy *create_rr(grpc_exec_ctx *exec_ctx,
   grpc_lb_policy_args args;
   memset(&args, 0, sizeof(args));
   args.client_channel_factory = glb_policy->cc_factory;
-  args.num_addresses = process_serverlist(serverlist, &args.lb_addresses);
-  args.user_data_vtable.copy = user_data_copy;
-  args.user_data_vtable.destroy = user_data_destroy;
+  const size_t num_ok_addresses =
+      process_serverlist(serverlist, &args.lb_addresses);
+  args.num_addresses = num_ok_addresses;
 
   grpc_lb_policy *rr = grpc_lb_policy_create(exec_ctx, "round_robin", &args);
+
+  glb_policy->num_ok_serverlist_addresses = num_ok_addresses;
+  if (glb_policy->lb_addresses != NULL) {
+    /* dispose of the previous version */
+    for (size_t i = 0; i < num_ok_addresses; ++i) {
```

Comment:
Shouldn't this use the original value of `glb_policy->num_ok_serverlist_addresses`, since the old and new sets of addresses might be different sizes?